core/constraint-streams/src/main/java/ai/timefold/solver/constraint/streams/bavet/common/GroupNodeConstructor.java

```diff
@@ -7,17 +7,138 @@
 import ai.timefold.solver.core.api.score.Score;
 import ai.timefold.solver.core.api.score.stream.ConstraintStream;
 import ai.timefold.solver.core.config.solver.EnvironmentMode;
+import ai.timefold.solver.core.impl.util.Pair;
+import ai.timefold.solver.core.impl.util.Quadruple;
+import ai.timefold.solver.core.impl.util.Triple;
 
 public interface GroupNodeConstructor<Tuple_ extends AbstractTuple> {
```

Comment:
It appears we no longer need the tuple here.
(Which also means we can now replace all declarations of `GroupNodeConstructor` variables with `var`.)

Actually, I checked the history and this was the case on `main` already.
Please make the changes anyway - it's a good thing to leave the code in a better shape than the one we found it in.

Reply:
It appears the `<Tuple_ extends AbstractTuple>` is actually an aid to the compiler; since it is used in the return type of the group by node constructors, it informs the compiler that `NodeConstructorWithAccumulate` uses `Tuple_` in the lambdas, allowing the `GroupNodeConstructorWithAccumulate` to accept it without a cast. Removing it makes a tons of casts necessary.  The alternative is to replace all `GroupNodeConstructor` usage with `GroupNodeConstructorWithAccumulate` or `GroupNodeConstructorWithoutAccumulate` (which does use the tuple type, in a constructor and a field).

Reply:
Interesting. IntelliJ shows it to me as unnecessary. Must be a bug then.

Reply:
IntelliJ is not wrong; it is unused in GroupNodeConstructor, but `GroupNodeConstructorWith(out)Accumulate` does this: `GroupNodeConstructorWith(out)Accumulate<Tuple_ extends AbstractTuple> implements GroupNodeConstructor<Tuple_>`, and then uses Tuple_ in a field and its constructor. So when you remove the `Tuple_` from `GroupNodeConstructor` in the return type, the compiler no longer knows the generic signature of `GroupNodeConstructorWith(out)Accumulate` to use, making a ton of casting required in GroupNodeConstructor and all the ConstraintStream(s).